src/main/java/hudson/plugins/s3/callable/S3GzipCallable.java

```diff
@@ -66,12 +66,17 @@ public void progressChanged(ProgressEvent event) {
     }
 
     @Override
-    @SuppressFBWarnings("RV_RETURN_VALUE_IGNORED_BAD_PRACTICE")
+    @SuppressFBWarnings({"RV_RETURN_VALUE_IGNORED_BAD_PRACTICE","OBL_UNSATISFIED_OBLIGATION"})
     public String invoke(FilePath file) throws IOException, InterruptedException {
         final File localFile = gzipFile(file);
         Upload upload = null;
 
-        try (final InputStream gzippedStream = new FileInputStream(localFile)) {
+        try {
+            // This stream is asynchronously used in startUploading,
+            // so we cannot use its AutoCloseable behaviour with a
+            // try-with-resources statement, as that would likely
+            // close the stream before the upload has succeeded.
+            final InputStream gzippedStream = new FileInputStream(localFile);
```

Comment:
Is startUploading closing it later?

Reply:
I do not know this with certainty, but as it is an async stream acceptor, I do believe that it at least should accept ownership of the stream, and thus close it when appropriate, and thus not be incorrect usage here. The current usage of the API is (AFAICS) no different from how it is used in the non-gzipped variant in https://github.com/jenkinsci/s3-plugin/blob/master/src/main/java/hudson/plugins/s3/callable/S3UploadCallable.java#L25

Reply:
I'm fine with reverting if the current behavior is not working, but I think it should be investigated whether a leak is happening and fix it.

Reply:
Seems like the stream is being closed after upload:

[Uploads.getInstance()](https://github.com/jenkinsci/s3-plugin/blob/master/src/main/java/hudson/plugins/s3/Uploads.java#L68-L76) returns an Uploads, which closes the [passed inputstream](https://github.com/jenkinsci/s3-plugin/blob/master/src/main/java/hudson/plugins/s3/Uploads.java#L24-L36) after completion [here](https://github.com/jenkinsci/s3-plugin/blob/master/src/main/java/hudson/plugins/s3/Uploads.java#L38-L50)

Unless my static analisys is incorrect, this stream will be closed once passed to `Uploads#startUploading`

Reply:
Great, thanks for taking the time to validate it!